File: client/app/models/template.js
// Template model class.
var Model = require("models/base/model");

module.exports = Model.extend({
    idAttribute: "_id"

  , defaults: {
      name: ""
    , template: ""
    , build: ""
    , regions: { header: "default", footer: "default" }
  }

  , setRegion: function (name, slug) {
    var regions = this.get("regions");
    regions[name] = slug;
    this.set("regions", regions);
  }
});

File: models/template.rb
class Template
  include Mongoid::Document

  field :name, type: String
  field :template, type: String
  field :regions, type: Hash, default: {header: 'default', footer: 'default'}

  embedded_in :theme

  def regions
    self[:regions].symbolize_keys
  end

  def slug
    self[:name].parameterize
  end

  # PHP filename
  def filename
    "#{self.slug}.php"
  end
end
